src/inmanta/agent/agent.py

```diff
@@ -1672,7 +1178,7 @@ async def ensure_code(self, code: Collection[ResourceInstallSpec]) -> FailedReso
             async with self._resource_loader_lock.get(resource_install_spec.resource_type):
                 # stop if the last successful load was this one
                 # The combination of the lock and this check causes the reloads to naturally 'batch up'
-                if self._last_loaded_version[resource_install_spec.resource_type] == resource_install_spec.model_version:
+                if self._last_loaded_version[resource_install_spec.resource_type] == resource_install_spec.blueprint:
```

Comment:
I don't see where the difference is? 

I.e. the only change I made here is packing arguments together into an object ?

Reply:
The difference is that equality of the packed object is less strict than equality of the model version (different model versions may, and often will, have the same blueprint). So I see how this is more efficient, but it may have slightly different loading behavior.

I mainly wanted to point it out to make sure you're aware of it, because you said it was important that nothing changed for the old executor flow. I think this qualifies as "close enough" though.